polymer/gulpfile.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+"use strict";
+
+const gulp = require('gulp-help')(require('gulp'));
+
+const gls = require('gulp-live-server');
+const uglify = require('gulp-uglify');
+const pump = require('pump');
+const cleanCSS = require('gulp-clean-css');
+const fs = require('fs');
+const gutil = require('gulp-util');
+const polymerBuild = require('polymer-build');
+const polymerJson = require('./polymer.json');
+const polymerProject = new polymerBuild.PolymerProject(polymerJson);
+const mergeStream = require('merge-stream');
+//var gulpServiceWorker = require('gulp-serviceworker');
+const addServiceWorker = polymerBuild.addServiceWorker;
+
+
+const paths = {
+  dist: {
+    dir: 'dist'
+  },
+  src: 'src'
+};
+
+gulp.task('build:polymer', 'Build the polymer app', function() {
+  const polymer = 'polymer/**/*'
+  let sourcesStream = polymerProject.sources()
+  let dependenciesStream = polymerProject.dependencies()
+  let buildStream = mergeStream(sourcesStream, dependenciesStream)
+  return buildStream.pipe(gulp.dest(paths.dist.dir));
+});
+
+gulp.task('build', 'Build the polymer app', [
+  'build:polymer',
+  'sw'
+]);
+
+function build() {
+  gutil.log('Build!');
+  return polymerBuild.addServiceWorker({
```

Comment:
Instead of a separate task for this, maybe take https://github.com/PolymerElements/generator-polymer-init-custom-build/blob/master/generators/app/gulpfile.js as inspiration. It has most of the stuff you want here, but with minification and optimization (and comments :smile: )

Reply:
I gave it a try but it doesn't work: when I run `gulp build:polymer` from the main `gulpfile` the service worker is not generated; while if I keep it into different tasks it is. Could you help?